lib/uksched/include/uk/tcb_impl.h

```diff
@@ -67,7 +67,7 @@ extern "C" {
  *   - (>=0): Success, tcb is initialized.
  *   - (<0): Negative error code, thread creation is canceled.
  */
-int uk_thread_uktcb_init(struct uk_thread *thread, void *tcb);
+int uk_thread_uktcb_init(struct uk_thread *thread, void *tcb) __nonnull;
```

Comment:
According to the comment, the thread is already created and initialized (`_uk_thread_struct_init_alloc`'s steps prior to calling `uk_thread_uktcb_init` ensure the thread will not point to NULL). Therefore, adding `__nonnull` is not necessary here